Flapjack/CoreData/CoreDataSource.swift

```diff
@@ -279,6 +292,32 @@ public class CoreDataSource<T: NSManagedObject & DataObject>: NSObject, NSFetche
     }
 
 
+    // MARK: Notification handlers
+
+    @objc
+    private func contextWasCreated(_ notification: Notification) {
+        guard isContextAZombie else { return }
+        guard let context = notification.object as? NSManagedObjectContext else { return }
+        isContextAZombie = false
+        controller = NSFetchedResultsController<NSManagedObject>(fetchRequest: fetchRequest, managedObjectContext: context, sectionNameKeyPath: sectionProperty, cacheName: cacheKey)
+        refetchIfNeeded()
+    }
+
+    @objc
+    private func contextWillBeDestroyed(_ notification: Notification) {
+        guard controller.managedObjectContext === notification.object as? NSManagedObjectContext else { return }
+
+        // Make sure our listener knows our objects are going away
+        if hasExecuted, let onChangeBlock = onChange, let fetchedObjects = fetchedObjects, !fetchedObjects.isEmpty {
+            let itemRemovals: [DataSourceChange] = fetchedObjects.compactMap { indexPath(for: $0) }.map { .delete(path: $0) }
+            let sectionRemovals: [DataSourceSectionChange] = (0..<numberOfSections).map { .delete(section: $0) }
+            onChangeBlock(itemRemovals, sectionRemovals)
+        }
```

Comment:
Is this really necessary?  I feel like by the time a context is being destroyed, who really cares about the objects?

Reply:
Data sources and their parent objects need to know they're going away in case the data sources / parent helpers don't get deleted in between database nukes (we have at least one in our code that I know of that listens for… live trainings maybe?).